controllers/sandbox_controller.go

```diff
@@ -76,22 +79,79 @@ func (r *SandboxReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, nil
 	}
 
+	oldStatus := sandbox.Status.DeepCopy()
+
 	// Create a hash from the sandbox.Name and use it as label value
 	nameHash := NameHash(sandbox.Name)
-	_, err := r.reconcilePod(ctx, sandbox, nameHash)
-	if err != nil {
-		return ctrl.Result{}, err
+	var conditions []*metav1.Condition
+	condition, err := r.reconcilePod(ctx, sandbox, nameHash)
+	conditions = append(conditions, condition)
+	if err == nil {
+		condition, err = r.reconcileService(ctx, sandbox, nameHash)
+		conditions = append(conditions, condition)
 	}
+	updateErr := r.updateStatus(ctx, oldStatus, sandbox, conditions)
+	return ctrl.Result{}, errors.Join(err, updateErr)
+}
 
-	_, err = r.reconcileService(ctx, sandbox, nameHash)
-	if err != nil {
-		return ctrl.Result{}, err
+func (r *SandboxReconciler) updateStatus(ctx context.Context, oldStatus *sandboxv1alpha1.SandboxStatus, sandbox *sandboxv1alpha1.Sandbox, conditions []*metav1.Condition) error {
+	log := log.FromContext(ctx)
+
+	falseConditionSeen := false
+	message := ""
+	conditionsCount := 0
+	for _, condition := range conditions {
+		if condition == nil {
+			continue
+		}
+		conditionsCount++
+		if condition.Status == metav1.ConditionFalse {
+			falseConditionSeen = true
+		}
+		if message == "" {
+			message = condition.Message
+		} else {
+			message += "; " + condition.Message
+		}
+	}
+
+	// If all conditions are present OR we see atleast one false condition we update the Ready
+	if falseConditionSeen || len(conditions) == conditionsCount {
```

Comment:
This is a bit hard to follow and this assumption could break easily when we introduce more conditions. Suggest keeping the part about returning Pod/Service, and add a function here to generate condition, so that all the condition related logic is in the same place.

Reply:
refactored to a simpler logic. 
1. reconcile pod
2. reconcile service
3. compute and set Ready
4. update status